container/compliance/generate-attributions-md/src/dynamo_attributions/extractor.py

```
@@ -0,0 +1,146 @@
+
+from .lockfile import parse_cargo_lock, parse_go_mod
+from .tree import DependencyTree
+from .types import Ecosystem, ResolvedPackage
+
+logger = logging.getLogger(__name__)
+
+
+def _git_read(dynamo_path: str, branch: str, file_path: str) -> str:
+    """Read file content from a git ref without checkout."""
+    try:
+        result = subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)